Publish raw SWV stream to TCP port

patrislav1 opened the pull request:
This patch implements a [raw SWO stream server](https://github.com/pyocd/pyOCD/issues/1009) for tools like [orbuculum](https://github.com/orbcode/orbuculum).

I introduced a new option `tpiu_port` that is set to 3443 by default. The TPIU raw server is started implicitly, when `enable_swv` is activated. (If this is not desirable, then the default could also be changed to zero and the user would have to set it explicitly to 3443 or whatever to enable it).

I tested it on a STM32 Nucleo board with `pyocd gdbserver --target stm32l433rctxp -O enable_swv=true -O swv_system_clock=16000000` and a `orbcat -c 1,"0x%08x\n"` on the other side.

Fixes #1009

flit commented on pyocd/trace/swv.py:
```diff
@@ -155,6 +157,10 @@ def run(self):
             self._lock.acquire()
 
+        tpiu_port = self._session.options.get('tpiu_port')
+        tpiu_server = StreamServer(tpiu_port, name="TPIU", is_read_only=True) \
+                      if tpiu_port else None
```
The other servers in pyocd support setting the port to 0, which causes the OS to select any available high-numbered port. The actual port is then read back and printed in the "server started" log message. To do this for the raw SWV server might require adding a separate `swv_raw_server_enable`.

flit reviewed (commented):
Awesome, thanks!

Starting the raw server by default is ok.

patrislav1 commented:
Thanks for the quick review. Updated the PR according to your suggestions and also added to the `options.md` (It looks a bit like it could have been auto generated from the options dict, however I didn't find generator code, so I updated it manually anyway).

By the way, there's still some additional magic that `openocd "tpiu config internal ..."` is doing to the target to enable SWO that pyocd is not doing yet. If I cold start the target and enable SWV from pyocd, then the STM32 gets stuck at `while (ITM->PORT[TRACE_CHANNEL].u32 == 0)` - if I run `openocd "tpiu ..."` then that is resolved.
Not sure if this SWO enable magic is something that is in the scope of pyocd or not, but I can imagine it creating some confusion if openocd handles it one way and pyocd handles it differently. WDYT?

40Grit commented:
Everything is finally coming together. The life of the embedded developer should soon be less frustrating.

flit commented:
Thanks. The `options.md` file is still manually updated at the moment (whereas `command_reference.md` is auto-generated).

Regarding the STM32, that's a good question. It probably should be handled by pyocd. There's a `trace_start()` method in `SoCTarget` that can be overridden. However, it's also a delegate method, so it can be implemented in a user script (`pyocd_user.py`). And I believe the STM32 CMSIS-Packs provide a [debug sequence](https://arm-software.github.io/CMSIS_5/Pack/html/debug_description.html) that enables the trace clock. Pyocd doesn't support debug sequences yet, but I'm [working on it](https://github.com/flit/pyOCD/tree/feature/debug_sequences) in the background. But handling this doesn't need to be part of this PR.

flit commented:
/morph test
